apps/aecore/lib/aecore/chain/block_validation.ex

```diff
@@ -134,13 +134,16 @@ defmodule Aecore.Chain.BlockValidation do
   end
 
   @spec build_merkle_tree(list(SignedTx.t())) :: tree()
-  def build_merkle_tree([]), do: <<0::256>>
-
   def build_merkle_tree(txs) do
-    Enum.reduce(txs, PatriciaMerkleTree.new(:txs), fn tx, trie ->
-      encoded_tx = tx.data |> DataTx.rlp_encode()
-      PatriciaMerkleTree.enter(trie, encoded_tx |> Hash.hash(), encoded_tx)
-    end)
+    build_merkle_tree(txs, 0, PatriciaMerkleTree.new(:txs))
+  end
+
+  defp build_merkle_tree([], _position, tree), do: tree
+
+  defp build_merkle_tree([signed_tx | list_txs], position, tree) do
```

Comment:
We should pattern match to make sure that the tree only contains SignedTx. For instance:
```
defp build_merkle_tree([%SignedTx{} = signed_tx, | list_txs], position, tree) do
```